src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -65,7 +76,8 @@ export const useTradeExecution = ({
     const supportsEIP1559 = supportsETH(wallet) && (await wallet.ethSupportsEIP1559())
 
     return new Promise<void>(async (resolve, reject) => {
-      const execution = new TradeExecution()
+      const execution: TradeExecutionBase =
+        swapperName === SwapperName.OneInch ? new TradeExecution2() : new TradeExecution()
```

Comment:
Do we want to TODO this guy? I trust you next swappers will follow up migrating to the same implementation, but my OCD is sad without a TODO in the meantime